OCast/Classes/Core/Device/DeviceDiscovery.swift

```diff
@@ -195,71 +188,46 @@ import CocoaAsyncSocket
         sendMSearch()
         mSearchTimer = Timer.scheduledTimer(timeInterval: mSearchTimeout, target: self, selector: #selector(mSearchTimerExpiry), userInfo: nil, repeats: true)
 
-        isRunning = true
-
         return true
     }
 
     /**
      Stops a discovery process.
      */
     @objc public func stop() {
-        ssdpSocket?.close()
-        mSearchTimer.invalidate()
-        isRunning = false
+        ssdpSocket.close()
     }
 
     // MARK: - UDP management
 
     /// :nodoc:
     public func udpSocket(_ udpSocket : GCDAsyncUdpSocket, didReceive data: Data, fromAddress address: Data, withFilterContext _: Any?) {
-        
-        guard let udpData: NSString = NSString(data: data, encoding: String.Encoding.utf8.rawValue) else {
-            return
-        }
-
-        //OCastLog.debug("Ok From \(udpSocket.localHost() ?? "") on port \(udpSocket.localPort())\n\(udpData)")
- 
-        guard let searchTarget = getStickSearchTarget(fromUDPData: udpData as String) else {
-            OCastLog.error("Search Target is missing or corrputed. Aborting.")
-            return
-        }
-
-        if !isTargetMatching(for: searchTarget) {
-            OCastLog.debug("Warning: Search Target is not as exepected \(searchTarget). Aborting.")
-            return
-        }
-
-        guard let location = getStickLocation(fromUDPData: udpData as String) else {
-            OCastLog.error("Location parameter is missing.")
-            return
+        guard let udpString = String(data: data, encoding: .utf8),
+            let searchTarget = getStickSearchTarget(fromUDPData: udpString),
+            isTargetMatching(for: searchTarget),
+            let location = getStickLocation(fromUDPData: udpString) else {
+                OCastLog.error("Bad SSDP response")
+                return
         }
 
         initiateHttpRequest(from: self, with: .get, to: location, onSuccess: didReceiveHttpResponse(response:with:), onError: { _ in })
     }
 
     /// :nodoc:
-    public final func udpSocketDidClose(_: GCDAsyncUdpSocket, withError _: Error?) {
-        OCastLog.debug("UDP Socket did close")
+    public final func udpSocketDidClose(_ sock: GCDAsyncUdpSocket, withError error: Error?) {
+        mSearchTimer.invalidate()
+        resetContext()
+        
+        delegate?.deviceDiscoveryDidStop(self, withError: error)
     }
 
     func sendMSearch() {
-        if let ssdpSocket = ssdpSocket {
-            var tagIndex = 0
-            mSearchTargets.forEach { (target) in
-                let mSearchString = "M-SEARCH * HTTP/1.1\r\nHOST: \(ssdpAddress):\(ssdpPort)\r\nMan: \"ssdp:discover\"\r\nMX:\(mSearchTimeout) \r\nST: \(target)\r\n\r\n"
+        var tagIndex = 0
+        mSearchTargets.forEach { (target) in
+            let mSearchString = "M-SEARCH * HTTP/1.1\r\nHOST: \(ssdpAddress):\(ssdpPort)\r\nMan: \"ssdp:discover\"\r\nMX:\(mSearchTimeout) \r\nST: \(target)\r\n\r\n"
```

Comment:
Even if case is ignored for HTTP headers, should be more like `MAN:` than `Man:`...